app/src/main/java/com/example/playlistmaker/SearchHistory.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.playlistmaker
+
+import android.content.SharedPreferences
+import com.example.playlistmaker.dataclasses.ITunesTrack
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+
+class SearchHistory(val sharedPreferences: SharedPreferences) {
+
+    private var trackList: MutableList<ITunesTrack> = mutableListOf()
```

Comment:
```suggestion
    private val trackList: MutableList<ITunesTrack> = mutableListOf()
```
Должно быть достаточно, что сам список мьютабельный, делать переменную мьютабельной дополнительно не требуется

Reply:
![Screenshot_2](https://github.com/tLen-tLen/PlaylistMaker/assets/56912408/aba9f035-00ab-4a85-aaf6-3cf297f0500c)

Студия подсвечивает ошибку так

Reply:
`trackList` вообще удалить можно

Reply:
из функции `write` не стоит возвращать список, но если сильно нужно то можно написать `return read()`
а от самой переменной `trackList` можно избавиться